### DISCUSSION TITLE
Used ports viewer TUI

### DISCUSSION THREAD

**Discussion by dlvhdr** [charmbracelet/inspo#16] [Ideas]
**Is your feature request related to a problem? Please describe.**
I would like to see which ports are used by processes, usually so I can kill them 😈
But also, so I know which "server" processes are running, and which ports are taken.

**Describe the solution you'd like**

* Could either be inline output like [duf](https://github.com/muesli/duf) or a full screen TUI.
* Ability to select a row and perform actions on it (e.g. restart, kill).

**Comment by allyring** [comment]
I've been wanting to pick up Go for a while because of the cool stuff Charm's doing with the CLI, and I might try taking this as a first proper project.

Also got a few questions on your ideas for implemention:
- What platform are you on? I've done a bit of experementation using WSL and turns out Windows and Unix handle some of my initial ideas on how to get the info on ports a little differently, so I'd rather just try and develop for one of the two initially.

- What data would you like shown in the output? You'd want the port and process name / ID of course, but is there anything else that might be helpful to have? Only things I could think of were maybe the remote address that you're connecting to.

- Any thoughts on the controls you'd like to have over each process? Killing the process is easy enough, but restarting it seems a tad finnicky based on initial googling. Another idea I had was perhaps adding a rule to your firewall to block that port in the future? Just an idea of course, but I don't think it would be too complicated (though correct me if I'm wrong).

That's more or less it, sorry for the rambling lol.

**Comment by allyring** [comment]
Made a prototype! https://github.com/allyring/pvw
